Frontend/src/pages/register/components/Processing.jsx

```diff
@@ -0,0 +1,48 @@
+import { useQuery } from '@tanstack/react-query'
+import React, { useState } from 'react'
+import { Message } from 'semantic-ui-react'
+import { pollRegistrations } from '../../../api/registration/get/poll_registrations'
+
+const REFETCH_INTERVAL = 3000
+
+export default function Processing({ processing, setProcessing }) {
+  const [pollCounter, setPollCounter] = useState(0)
+  const { data } = useQuery({
+    queryKey: ['registration-status-polling'],
+    queryFn: async () => pollRegistrations(),
+    refetchInterval: REFETCH_INTERVAL,
+    enabled: processing,
+    onSuccess: () => {
+      setPollCounter(pollCounter + 1)
+    },
+  })
+  if (
+    data &&
+    (data.status.payment === 'initialized' ||
+      data.status.competing === 'incoming')
+  ) {
+    setProcessing(false)
+  }
+  return (
+    processing && (
+      <div>
+        <div>Your registration is processing...</div>
+        <div>{data && `Registration status: ${data.status.competing}`}</div>
+        <div>
```

Comment:
Did you mean to have the `{data &&` type stuff outside each `div`, or is it important that the `div`s show up even if they are empty? (I don't have this running locally so not sure exactly what everything looks like.)

Reply:
I was doing this so I can already start blurring the background except for the Div to indicate that something is processing.